File: services/business-services/business-menu-option-v3/application/service/menu_option_service.go
package service

import (
	"business-menu-option-v3/domain/model/request"
	"business-menu-option-v3/domain/model/response"
)

type MenuOptionService interface {
	FindAll() ([]response.MenuOptionResponse, error)
	FindById(id uint) (*response.MenuOptionResponse, error)
	Save(request request.MenuOptionRequest) (*response.MenuOptionResponse, error)
	Update(request request.MenuOptionRequest, id uint) (*response.MenuOptionResponse, error)
	Delete(id uint) error
}

File: services/business-services/menu-v2/src/main/java/com/demo/bbq/business/menuoption/application/service/MenuOptionService.java
package com.demo.bbq.business.menuoption.application.service;

import com.demo.bbq.business.menuoption.domain.model.request.MenuOptionSaveRequest;
import com.demo.bbq.business.menuoption.domain.model.request.MenuOptionUpdateRequest;
import com.demo.bbq.business.menuoption.domain.model.response.MenuOption;
import io.smallrye.mutiny.Uni;
import java.util.List;

public interface MenuOptionService {

  Uni<List<MenuOption>> findByCategory(String categoryCode);

  Uni<MenuOption> findByProductCode(String productCode);

  Uni<Void> save(MenuOptionSaveRequest menuOptionRequest);

  Uni<Void> update(MenuOptionUpdateRequest menuOptionRequest, String productCode);

  Uni<Void> deleteByProductCode(String productCode);
}
